jac-uk/admin
# #1863 Merit list report

## What's included?
The merit list report collates scores from the key tasks carried out during processing applications.

It includes the following features:
- [x] Search. Searches reference number and candidate name
- [x] Sort. Candidate name
- [x] Expand/collapse
- [x] Row & column header freeze
- [x] Scrolling
- [x] Full-screen view
- [x] Calculates total scores
- [x] Includes diversity data

Other tasks:
- [x] Show candidate name instead of ref number. Column title change to 'Candidate'
- [x] Open application link in new tab
- [x] Remove export button
- [x] Format Numbers
- [x] ~Look at making it editable~ Moved to [backwards compatibility](https://github.com/jac-uk/admin/issues/1883)
- [x] ~Look at copy and paste~ Moved to [backwards compatibility](https://github.com/jac-uk/admin/issues/1883)

Currently we are pulling data from the following tasks:
 - [x] Critical Analysis Test
 - [x] Situational Judgement Test
 - [x] Scenario Test

## Who should test?
✅ Product owner
✅ UTG

## How to test?
Locate an exercise which has completed Qualifying Tests, Scenario Test or Sift using the new Tasks.

Go to Reports > Merit list

Check that you can see data on the merit list report and perform the following basic functions:
 - Search for a particular candidate
 - Sort by candidate name or total score
 - Expand/collapse the scores for QT, Scenarios and/or Sift
 - Open and close Full screen view
 - Scroll the table contents and still see row & column headers (if there's enough data for scrolling to be needed)


## Risk - how likely is this to impact other areas?
🟠 Medium risk - this does change code that is shared with other areas

## Additional context
<img width="802" alt="image" src="https://user-images.githubusercontent.com/8524401/218754149-57dcec8f-1386-4802-a654-2781c6855913.png">


## Related permissions
Have permissions been considered for this functionality?
- No permission changes required

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_
